brainscore_vision/nwb_integration/create_assembly.py

```diff
@@ -0,0 +1,203 @@
+    return neuroid_meta
+
+def get_QC_neurids(nwb_file: NWBFile) -> Tuple[np.array, np.array]:
+    '''
+    This Method uses logical OR to find the common QC channels. (Closer to the BrainScore Method)
+    '''
+    psth = nwb_file.scratch['PSTHs_ZScored_SessionMerged'][:]
+    common_QC_channels = np.logical_and.reduce(nwb_file.scratch['QualityApprovedChannelMasks'])
+    channel_masks_day = nwb_file.scratch['QualityApprovedChannelMasks'][:]
+    channel_mask_all_list = []
+    day = 0
+    for key in sorted(nwb_file.scratch.keys()):
+        if key.startswith('PSTHs_QualityApproved_20'):
+            nreps_per_day = nwb_file.scratch[key][:].shape[1]
+            for i in range(nreps_per_day):
+                channel_mask_all_list.append(channel_masks_day[day,:])
+            day += 1
+    channel_mask_all = np.array(channel_mask_all_list)
+
+    assert channel_mask_all.shape[0] == psth.shape[1]
+    filtered_neurids = np.any(channel_mask_all, axis=0)
+    return filtered_neurids, common_QC_channels
+
+def filter_neuroids(assembly: NeuronRecordingAssembly, threshold: float) -> NeuronRecordingAssembly:
+    ceiler = load_ceiling('internal_consistency')
+    ceiling = ceiler(assembly)
+    ceiling = ceiling.raw
+    ceiling = CrossValidation().aggregate(ceiling)
+    pass_threshold = ceiling >= threshold
+    assembly = assembly[{'neuroid': pass_threshold}]
+    return assembly
+
+def load_responses(nwb_file: NWBFile, json_file_path: str, 
+                   stimuli: StimulusSet, use_QC_data: bool = True, 
+                   do_filter_neuroids: bool = False, 
+                   use_brainscore_filter_neuroids_method: bool = False) -> NeuronRecordingAssembly:
+    #-----------------------------------------------------------------------------------------------------------------------------
+    # Get the PSTH and normalizer PSTH
+    #-----------------------------------------------------------------------------------------------------------------------------
+    normalizer_psth = nwb_file.scratch['PSTHs_Normalizers_SessionMerged'][:]
+    if not use_QC_data:
+        psth            = nwb_file.scratch['PSTHs_ZScored_SessionMerged'][:]
+    elif use_QC_data:
+        psth            = nwb_file.scratch['PSTHs_QualityApproved_ZScored_SessionMerged'][:]
+    with open(json_file_path, 'r') as f:
+        params = json.load(f)
+    meta = np.array([params['start_time_ms'], params['stop_time_ms'], params['tb_ms']], dtype=int)
+    subject = params['subject']
+    qc_array_or, qc_array_and = get_QC_neurids(nwb_file)
+    #-----------------------------------------------------------------------------------------------------------------------------
+    # Compute firing rates.
+    #-----------------------------------------------------------------------------------------------------------------------------
+    timebase = np.arange(meta[0], meta[1], meta[2])
+    timebins = np.asarray([[int(x), int(x)+int(meta[2])] for x in timebase])
```

Comment:
If you are confident in the logic to grab `timebins`, then I would just add a check here again the user-submitted timelines to make sure they are the same